src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -563,6 +583,9 @@ public static IAsyncResult BeginGetHostEntry(string hostNameOrAddress, AsyncCall
         public static IAsyncResult BeginGetHostEntry(IPAddress address, AsyncCallback requestCallback, object stateObject)
         {
             if (NetEventSource.Log.IsEnabled()) NetEventSource.Enter(NetEventSource.ComponentType.Socket, "DNS", "BeginGetHostEntry", address);
+
+            NameResolutionPal.EnsureSocketsAreInitialized();
```

Comment:
Looks like the XxAsync methods that wrap these Begin/End methods also call these methods.  Since they just delegate here, can we remove those calls from there?